packages/BetterPhpDocParser/Annotation/ChangeResolvedClassInParticularContextForAnnotation.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\BetterPhpDocParser\Annotation;
+
+use Rector\BetterPhpDocParser\PhpDoc\DoctrineAnnotationTagValueNode;
+use Rector\BetterPhpDocParser\ValueObject\PhpDoc\DoctrineAnnotation\CurlyListNode;
+use Rector\Php80\ValueObject\AnnotationToAttribute;
+
+class ChangeResolvedClassInParticularContextForAnnotation
+{
+    private const RULES = [
+        [
+            'tag' => 'Doctrine\ORM\Mapping\JoinTable',
+            'value' => 'inverseJoinColumns',
+            'resolved_class' => 'Doctrine\ORM\Mapping\InverseJoinColumns',
+        ],
```

Comment:
Arrays with nested strings are quite hard to operate with and type check. E.g. PHPStan sees this as `mixed[]` and could not verify the validity.

Could you use value object with 3 parameters to make this stricter?

Reply:
done

Reply:
DIC try to autowire the valueObject, how to tell him not to do it ?

```
annot autowire service "Rector\BetterPhpDocParser\Annotation\ValueObject\ChangeResolvedClassInParticularContextForAnnotationRule": argument "$tag" of method "__construct()" is type-hinted "string", you should configure its value explicitly
```

Reply:
It should be excluded in `services.php-packages`:  https://github.com/rectorphp/rector-src/blob/127428ac8ab5d0e360cadd79bb69e14e0d5636b3/config/services-packages.php#L17

Reply:
done